diff --git a/src/Entity/Entity.hpp b/src/Entity/Entity.hpp
index 9602e92..f03cfa0 100644
--- a/src/Entity/Entity.hpp
+++ b/src/Entity/Entity.hpp
@@ -52,7 +52,7 @@ class Entity {
     template <typename T>
     const T& GetComponent() const {
         BE_ASSERT(HasComponent<T>());
-        return m_Registry->get<T>(m_Handle);
+        return std::as_const(m_Registry)->get<T>(m_Handle);
     }
 
     bool IsNull() const {
diff --git a/src/Entity/EntityBehavior.cpp b/src/Entity/EntityBehavior.cpp
index 20e533a..42ab247 100644
--- a/src/Entity/EntityBehavior.cpp
+++ b/src/Entity/EntityBehavior.cpp
@@ -23,7 +23,7 @@ void EntityBehavior::Destroy(Entity e) {
 }
 
 const Components::Transform& EntityBehavior::Transform() const {
-    return GetComponent<Components::Transform>();
+    return std::as_const(*this).GetComponent<Components::Transform>();
 }
 
 Components::Transform& EntityBehavior::Transform() {
@@ -35,7 +35,7 @@ Components::Tag& EntityBehavior::Tag() {
 }
 
 const Components::Tag& EntityBehavior::Tag() const {
-    return GetComponent<Components::Tag>();
+    return std::as_const(*this).GetComponent<Components::Tag>();
 }
 
 const UUID EntityBehavior::ID() const {
diff --git a/src/Entity/EntityBehavior.hpp b/src/Entity/EntityBehavior.hpp
index 88d63d1..1a5f729 100644
--- a/src/Entity/EntityBehavior.hpp
+++ b/src/Entity/EntityBehavior.hpp
@@ -88,7 +88,7 @@ class EntityBehavior {
      */
     template <typename T>
     const inline T& GetComponent() const {
-        return m_Entity.GetComponent<T>();
+        return std::as_const(m_Entity).GetComponent<T>();
     }
 
     /**
